### PR TITLE
Remove pipenv

### PR DESCRIPTION
**Pull request by mmucek95**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -62,12 +56,14 @@ COPY examples/__init__.py ./examples
 
 COPY wca ./wca
 
-ENTRYPOINT ["pipenv", "run", "python3.6", "wca/main.py"]
+ENTRYPOINT ["evn/bin/python", "wca/main.py"]
```
Comment (Damenus) [inline Dockerfile]: Typo

File: Makefile [hunk 2]
```diff
@@ -36,40 +39,59 @@ endif
...
+ACTIV = env/bin/activate
+
 venv:
-	@echo Preparing virtual enviornment using pipenv.
-	pipenv --version
-	env PIPENV_QUIET=true pipenv install --dev
+	@echo Preparing virtual enviornment.
+	python3.6 -m venv env
+	source $(ACTIV) && \
+	pip install -r requirements.txt && \
```
Comment (Creatone) [inline Makefile]: You can put it in a function like ``execute_in_venv( command )``.

File: Makefile [hunk 3]
```diff
@@ -36,40 +39,59 @@ endif
...
+	deactivate
 
 flake8:
 	@echo Checking code quality.
-	pipenv run flake8 wca tests examples
+	source $(ACTIV) && \
+	flake8 wca tests examples && \
```
Comment (Creatone) [inline Makefile]: You can put it in a function like ``execute_in_venv( command )``.

File: Makefile [hunk 4]
```diff
@@ -36,40 +39,59 @@ endif
...
+	deactivate
 
 bandit:
 	@echo Checking code with bandit.
-	pipenv run bandit -r wca -s B101 -f html -o wca-bandit.html
+	source $(ACTIV) && \
+	bandit -r wca -s B101 -f html -o wca-bandit.html && \
```
Comment (Creatone) [inline Makefile]: You can put it in a function like ``execute_in_venv( command )``.

File: Makefile [hunk 5]
```diff
@@ -36,40 +39,59 @@ endif
...
+	deactivate
 
 bandit_pex:
 	@echo Checking pex with bandit.
 	unzip dist/wca.pex -d dist/wca-pex-bandit
-	pipenv run bandit -r dist/wca-pex-bandit/.deps -s B101 -f html -o wca-pex-bandit.html || true
-	rm -rf dist/wca-pex-bandit
+	source $(ACTIV) && \
+	bandit -r dist/wca-pex-bandit/.deps -s B101 -f html -o wca-pex-bandit.html || true && \
```
Comment (Creatone) [inline Makefile]: You can put it in a function like ``execute_in_venv( command )``.

File: Makefile [hunk 6]
```diff
@@ -36,40 +39,59 @@ endif
...
+	deactivate
+
+	rm -rf dist/wca-pex-bandit 
 
 check_outdated:
 	@echo Checking out of date dependencies.
-	pipenv run pip list --outdated
+	source $(ACTIV) && \
+	pip list --outdated && \
```
Comment (Creatone) [inline Makefile]: You can put it in a function like ``execute_in_venv( command )``.

File: Makefile [hunk 7]
```diff
@@ -36,40 +39,59 @@ endif
...
+	deactivate
 
 unit:
 	@echo Running unit tests.
-	pipenv run env PYTHONPATH=.:examples/workloads/wrapper pytest --cov-report term-missing --cov=wca tests --ignore=tests/e2e/test_wca_metrics.py
+	source $(ACTIV) && \
+	env PYTHONPATH=.:examples/workloads/wrapper pytest --cov-report term-missing --cov=wca tests --ignore=tests/e2e/test_wca_metrics.py && \
```
Comment (Creatone) [inline Makefile]: You can put it in a function like ``execute_in_venv( command )``.

File: Makefile [hunk 8]
```diff
@@ -36,40 +39,59 @@ endif
...
+	deactivate
 
 unit_no_ssl:
 	@echo Running unit tests.
-	pipenv run env PYTHONPATH=.:examples/workloads/wrapper pytest --cov-report term-missing --cov=wca tests --ignore=tests/e2e/test_wca_metrics.py --ignore=tests/ssl/test_ssl.py
+	source $(ACTIV) && \
+	run env PYTHONPATH=.:examples/workloads/wrapper pytest --cov-report term-missing --cov=wca tests --ignore=tests/e2e/test_wca_metrics.py --ignore=tests/ssl/test_ssl.py && \
```
Comment (Creatone) [inline Makefile]: You can put it in a function like ``execute_in_venv( command )``.

File: Makefile [hunk 9]
```diff
@@ -36,40 +39,59 @@ endif
...
+	deactivate
 
 junit:
-	@echo Running unit tests.	
-	pipenv run env PYTHONPATH=.:examples/workloads/wrapper pytest --cov-report term-missing --cov=wca tests --junitxml=unit_results.xml -vvv -s --ignore=tests/e2e/test_wca_metrics.py
+	@echo Running unit tests.
+	source $(ACTIV) && \
+	env PYTHONPATH=.:examples/workloads/wrapper pytest --cov-report term-missing --cov=wca tests --junitxml=unit_results.xml -vvv -s --ignore=tests/e2e/test_wca_metrics.py && \
```
Comment (Creatone) [inline Makefile]: You can put it in a function like ``execute_in_venv( command )``.

File: Makefile [hunk 10]
```diff
@@ -106,11 +128,13 @@ wca_package_in_docker_with_kafka:
...
 	@echo Building wca pex file.
-	-rm .pex-build/wca*
+	-rm -f .pex-build/wca*
 	-rm -rf .pex-build
-	-rm dist/wca.pex
+	-rm -f dist/wca.pex
 	-rm -rf wca.egg-info
-	pipenv run env PYTHONPATH=. pex . $(OPTIONAL_MODULES) $(PEX_OPTIONS) -o dist/wca.pex -m wca.main:main
+	source $(ACTIV) && \
+ 	pex . $(OPTIONAL_MODULES) $(PEX_OPTIONS) -o dist/wca.pex -m wca.main:main && \
```
Comment (Creatone) [inline Makefile]: You can put it in a function like ``execute_in_venv( command )``.

File: Makefile [hunk 11]
```diff
@@ -134,17 +158,20 @@ wrapper_package:
...
-	pipenv run pex . $(OPTIONAL_MODULES) -D examples/workloads/wrapper $(PEX_OPTIONS) -o dist/rpc_perf_wrapper.pex -m wrapper.parser_rpc_perf
-	pipenv run pex . $(OPTIONAL_MODULES) -D examples/workloads/wrapper $(PEX_OPTIONS) -o dist/mutilate_wrapper.pex -m wrapper.parser_mutilate
-	pipenv run pex . $(OPTIONAL_MODULES) -D examples/workloads/wrapper $(PEX_OPTIONS) -o dist/cassandra_stress_wrapper.pex -m wrapper.parser_cassandra_stress
-	pipenv run pex . $(OPTIONAL_MODULES) -D examples/workloads/wrapper $(PEX_OPTIONS) -o dist/stress_ng_wrapper.pex -m wrapper.parser_stress_ng
-	pipenv run pex . $(OPTIONAL_MODULES) -D examples/workloads/wrapper $(PEX_OPTIONS) -o dist/memtier_benchmark_wrapper.pex -m wrapper.parser_memtier
-	pipenv run pex . $(OPTIONAL_MODULES) -D examples/workloads/wrapper $(PEX_OPTIONS) -o dist/mysql_tpm_gauge_wrapper.pex -m wrapper.parser_mysql_tpm_gauge
-	./dist/wrapper.pex --help >/dev/null
+
+	source $(ACTIV) && \
+    pex . $(OPTIONAL_MODULES) -D examples/workloads/wrapper $(PEX_OPTIONS) -o dist/wrapper.pex -m wrapper.wrapper_main && \
```
Comment (Creatone) [inline Makefile]: You can put it in a function like ``execute_in_venv( command )``.

File: Makefile [hunk 12]
```diff
@@ -165,16 +192,19 @@ _unsafe_wrapper_package:
...
-	pipenv run $(ENV_UNSAFE) pex . -D examples/workloads/wrapper $(PEX_OPTIONS) -o dist/ycsb_wrapper.pex -m wrapper.parser_ycsb
-	pipenv run $(ENV_UNSAFE) pex . -D examples/workloads/wrapper $(PEX_OPTIONS) -o dist/rpc_perf_wrapper.pex -m wrapper.parser_rpc_perf
-	pipenv run $(ENV_UNSAFE) pex . -D examples/workloads/wrapper $(PEX_OPTIONS) -o dist/mutilate_wrapper.pex -m wrapper.parser_mutilate
-	pipenv run $(ENV_UNSAFE) pex . -D examples/workloads/wrapper $(PEX_OPTIONS) -o dist/cassandra_stress_wrapper.pex -m wrapper.parser_cassandra_stress
-	pipenv run $(ENV_UNSAFE) pex . -D examples/workloads/wrapper $(PEX_OPTIONS) -o dist/stress_ng_wrapper.pex -m wrapper.parser_stress_ng
-	pipenv run $(ENV_UNSAFE) pex . -D examples/workloads/wrapper $(PEX_OPTIONS) -o dist/memtier_benchmark_wrapper.pex -m wrapper.parser_memtier
-	./dist/wrapper.pex --help >/dev/null
+
+	source $(ACTIV) && \
+	$(ENV_UNSAFE) pex . -D examples/workloads/wrapper $(PEX_OPTIONS) -o dist/wrapper.pex -m wrapper.wrapper_main && \
```
Comment (Creatone) [inline Makefile]: You can put it in a function like ``execute_in_venv( command )``.

File: Makefile [hunk 13]
```diff
@@ -195,4 +225,6 @@ tester:
 
 generate_docs:
 	@echo Generate documentation.
-	pipenv run env PYTHONPATH=. python util/docs.py
+	source $(ACTIV) && \
+ 	env PYTHONPATH=. python util/docs.py && \
```
Comment (Creatone) [inline Makefile]: You can put it in a function like ``execute_in_venv( command )``.

**Pull request merged**